deel/torchlip/utils/lconv_norm.py

```diff
@@ -24,24 +24,54 @@
 # rights reserved. DEEL is a research program operated by IVADO, IRT Saint Exupéry,
 # CRIAQ and ANITI - https://www.deel.ai/
 # =====================================================================================
-from typing import Tuple
+from typing import Tuple, Union
 
 import numpy as np
 import torch
 import torch.nn as nn
 import torch.nn.utils.parametrize as parametrize
 
 
+def compute_lconv_coef_1d(
+    kernel_size: Tuple[int],
+    input_shape: Tuple[int] = None,
+    strides: Tuple[int] = (1,),
+    padding_mode: str = "zeros",
+) -> float:
+    stride = strides[0]
+    k1 = kernel_size[0]
+
+    if (
+        (padding_mode in ["zeros", "same"])
+        and (stride == 1)
+        and (input_shape is not None)
+    ):
+        # See https://arxiv.org/abs/2006.06520
+        in_l = input_shape[-1]
+        k1_div2 = (k1 - 1) / 2
+        coefLip = in_l / (k1 * in_l - k1_div2 * (k1_div2 + 1))
+    else:
+        sn1 = strides[0]
+        coefLip = 1.0 / np.ceil(k1 / sn1)
+
+    return coefLip  # type: ignore
+
+
 def compute_lconv_coef(
     kernel_size: Tuple[int, ...],
     input_shape: Tuple[int, ...] = None,
     strides: Tuple[int, ...] = (1, 1),
+    padding_mode: str = "zeros",
 ) -> float:
     # See https://arxiv.org/abs/2006.06520
     stride = np.prod(strides)
     k1, k2 = kernel_size
 
-    if stride == 1 and input_shape is not None:
+    if (
+        (padding_mode in ["zeros", "same"])
```

Comment:
see previous comment

Reply:
accepted in f3f32bbf560b48d3e96af79e931d0951afc710db